src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannelInsertionException.java

```diff
@@ -0,0 +1,15 @@
+package com.snowflake.kafka.connector.internal.streaming;
+
+/**
+ * Class for exceptions that occur while interacting with Snowflake through Snowpipe Streaming.
```

Comment:
`SFException` is not what I wanted to throw because we reopen the channel on that. I couldnt through ConnectException either since it would fail the task, hence I decided to create another RTE instance which could be more suitable for insertRows failures which can help us reset offset + get desired offset in Kafka. 

I used `RetriableException` before but that was incorrect. https://github.com/snowflakedb/snowflake-kafka-connector/pull/423/files#r848941980